services/api-server/src/simcore_service_api_server/api/routes/files.py

```diff
@@ -76,36 +156,88 @@ async def save_file(file):
 
 
 @router.get("/{file_id}", response_model=FileMetadata)
-async def get_file(file_id: UUID):
+async def get_file(
+    file_id: UUID,
+    storage_client: StorageApi = Depends(get_api_client(StorageApi)),
+    user_id: int = Depends(get_current_user_id),
+):
     """ Gets metadata for a given file resource """
+
     try:
-        return the_fake_impl.files[file_id]
-    except KeyError as err:
-        raise HTTPException(
-            status.HTTP_404_NOT_FOUND,
-            detail=f"File with identifier {file_id} not found",
-        ) from err
+        stored_files: List[StorageFileMetaData] = await storage_client.search_files(
+            user_id, file_id
+        )
+        if not stored_files:
+            raise ValueError("Not found in storage")
 
+        stored_file_meta = stored_files[0]
+        assert stored_file_meta.user_id == user_id  # nosec
+        assert stored_file_meta.file_id  # nosec
 
-@router.post("/{file_id}:download")
-async def download_file(file_id: UUID):
-    # TODO: hash or UUID? Ideally like container ids
-    try:
-        metadata = the_fake_impl.files[file_id]
-        file_path = the_fake_impl.get_storage_path(metadata)
+        # Adapts storage API model to API model
+        meta = convert_metadata(stored_file_meta)
+        return meta
 
-        return FileResponse(
-            str(file_path),
-            media_type=metadata.content_type,
-            filename=metadata.filename,
-        )
-    except KeyError as err:
+    except (ValueError, ValidationError) as err:
+        logger.debug("File %d not found: %s", file_id, err)
         raise HTTPException(
             status.HTTP_404_NOT_FOUND,
             detail=f"File with identifier {file_id} not found",
         ) from err
 
 
+@router.get("/{file_id}/content")
+async def download_file(
+    file_id: UUID,
+    storage_client: StorageApi = Depends(get_api_client(StorageApi)),
+    user_id: int = Depends(get_current_user_id),
+):
+    # gets meta
+    meta: FileMetadata = await get_file(file_id, storage_client, user_id)
+
+    # download from S3 using pre-signed link
+    presigned_download_link = await storage_client.get_download_link(
+        user_id, meta.file_id, meta.filename
+    )
+
+    logger.info("Downloading %s to %s ...", meta, presigned_download_link)
+
+    async def _download_chunk():
+        async with httpx.AsyncClient(timeout=httpx.Timeout(5.0, read=3600)) as client:
```

Comment:
I would put the read on an env var as well

Reply:
But I followed [the same as here](https://github.com/ITISFoundation/osparc-simcore/blob/feaa2e243a96ef8174dd85f42ad1b5a0c3d7f9fb/services/web/server/src/simcore_service_webserver/exporter/formatters/formatter_v1.py#L34)